.vib/magento/cypress/cypress/integration/magento_spec.js

```diff
@@ -0,0 +1,103 @@
+/// <reference types="cypress" />
+import { random, allowDataUsage } from './utils';
+
+before(() => {
```

Comment:
I'm not sure how this `before` works as you are always logging in but then in some tests you call `cy.login()` again.  Does this generate any type of conflict?

Reply:
I've added a `logout()` to the `before()` just to have a more uniform state.
Other than that, here are some things worth knowing: 

There are 2 types of users for this app

- Customer (no need for login)
- Admin (there is a need for login)

I am not doing the login in the customer tests since it is not needed.

The `before()` hook runs only once before all tests. 
I need a login in the `before()` since I need to allow data usage once before all the tests. In order to do that, the user requires to be logged in.
Hence, before all tests, I log in, allow data usage, and log out.
Then each test individually does or doesn't log in, depending if it's a customer or admin test.